sklearn/preprocessing/_encoders.py

```diff
@@ -0,0 +1,888 @@
+        Feature ``i`` in the original data is mapped to features
+        from ``feature_indices_[i]`` to ``feature_indices_[i+1]``
+        (and then potentially masked by `active_features_` afterwards)
+
+        .. deprecated:: 0.20
+
+    n_values_ : array of shape (n_features,)
+        Maximum number of values per feature.
+
+        .. deprecated:: 0.20
+
+    Examples
+    --------
+    Given a dataset with two features, we let the encoder find the unique
+    values per feature and transform the data to a binary one-hot encoding.
+
+    >>> from sklearn.preprocessing import OneHotEncoder
+    >>> enc = OneHotEncoder(handle_unknown='ignore')
+    >>> X = [['Male', 1], ['Female', 3], ['Female', 2]]
+    >>> enc.fit(X)
+    ... # doctest: +ELLIPSIS
+    OneHotEncoder(categories='auto', dtype=<... 'numpy.float64'>,
+           handle_unknown='ignore', sparse=True)
+
+    >>> enc.categories_
+    [array(['Female', 'Male'], dtype=object), array([1, 2, 3], dtype=object)]
+    >>> enc.transform([['Female', 1], ['Male', 4]]).toarray()
+    array([[ 1., 0., 1., 0., 0.],
+           [ 0., 1., 0., 0., 0.]])
+    >>> enc.inverse_transform([[0, 1, 1, 0, 0], [0, 0, 0, 1, 0]])
+    array([['Male', 1],
+           [None, 2]], dtype=object)
+
+    See also
+    --------
+    sklearn.preprocessing.OrdinalEncoder : performs an ordinal (integer)
+      encoding of the categorical features.
+    sklearn.feature_extraction.DictVectorizer : performs a one-hot encoding of
+      dictionary items (also handles string-valued features).
+    sklearn.feature_extraction.FeatureHasher : performs an approximate one-hot
+      encoding of dictionary items or strings.
+    sklearn.preprocessing.LabelBinarizer : binarizes labels in a one-vs-all
+      fashion.
+    sklearn.preprocessing.MultiLabelBinarizer : transforms between iterable of
+      iterables and a multilabel format, e.g. a (samples x classes) binary
+      matrix indicating the presence of a class label.
+    """
+
+    def __init__(self, n_values=None, categorical_features=None,
+                 categories=None, sparse=True, dtype=np.float64,
+                 handle_unknown='error'):
+        self._categories = categories
+        if categories is None:
+            self.categories = 'auto'
+        else:
+            self.categories = categories
+        self.sparse = sparse
+        self.dtype = dtype
+        self.handle_unknown = handle_unknown
+
+        if n_values is not None:
+            pass
+            # warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            n_values = "auto"
+        self._deprecated_n_values = n_values
+
+        if categorical_features is not None:
+            pass
+            # warnings.warn("Deprecated", DeprecationWarning)
+        else:
+            categorical_features = "all"
+        self._deprecated_categorical_features = categorical_features
+
+    # Deprecated keywords
+
+    @property
+    def n_values(self):
```

Comment:
Doesn't the ``deprecate`` decorator do this deprecation warning? Also all warnings need version info.

Reply:
I am not fully sure `deprecated` decorator will work with a setter, but need to check (was not aware of this utility)

Reply:
Regarding the version info in the warnings: will add that (I wanted to ask if there are some contributor guidelines around what I need to put in it, but I found them myself :-) http://scikit-learn.org/stable/developers/contributing.html#deprecation)